pkg/directory/types/node.go

```diff
@@ -190,10 +190,16 @@ func NodeCreate(ctx context.Context, db *mongo.Database, node Node) (schema.ID,
 		}
 		node.Created = schema.Date{Time: time.Now()}
 	} else {
+		// node exists, only update new fields
 		id = current.ID
-		// make sure we do NOT overwrite these field
-		node.Created = current.Created
-		node.FreeToUse = current.FreeToUse
+
+		tmp := node
+		node = current
```

Comment:
don't we also miss the public key here ?
The code that register the node is:
```go
err = store.NodeRegister(directory.Node{
		NodeId:    nodeID.Identity(),			NodeId:    nodeID.Identity(),
		NodeIdV1:  v1ID,			NodeIdV1:  v1ID,
		FarmId:    int64(farmID),			FarmId:    int64(farmID),
		OsVersion: version,			OsVersion: version,
		Location: directory.Location{			Location: directory.Location{
			Continent: loc.Continent,				Continent: loc.Continent,
			Country:   loc.Country,				Country:   loc.Country,
			City:      loc.City,				City:      loc.City,
			Longitude: loc.Longitute,				Longitude: loc.Longitute,
			Latitude:  loc.Latitude,				Latitude:  loc.Latitude,
		},			},
		PublicKeyHex: publicKeyHex,			PublicKeyHex: publicKeyHex,
		Uptime:       int64(uptime),
	})		})
```

Reply:
I thought it was immutable for a given node :)

Reply:
Actually this is indeed the case. Seems we should even check the signature of the request at this point and make sure that this is actually the node itself sending data...

Reply:
hmm yeah the problem is this handler is not authenticated. And it can't be since that would break things for totally new nodes I think.